src/tests/test_clang_bug.cpp

```diff
@@ -0,0 +1,125 @@
+      "B7XNGgV19v0enkDVawD8wbm+tl0yMiwil0FtQ4njwdDpyVHIfO/TxLTMYaoJz1Oo"
+      "VnjvsthPvNTZxzrXUJPS7YR2aIUwZMWJPRFny0NuEEf2Lqal+FLOaNZzLrsiP/xX"
+      "4WlXEf7t+K9UynjcJktJtXRl1n+63uQoPrn96/H6ohMW5ervjL+rbDOHcT4OK3oI"
+      "Dl4XDoedcxeO8DsQUPKNh10GyDkNjBDWhKrfyaON3vk6S6ooVffVKs1sU6ZImMpj"
+      "SrgtViqXoQh5YhNew2hO90dqgoNWXdxrWM51Qaf4l2NIwvyGSfslm6XgMvfrI4LN"
+      "++RvHRseGFa169cp0GY8kI8RHsYvo2ynSOpi84E9OeX8HojNim1xcCieZwTr/CXi"
+      "EbvPD/Y+c6OCIKDJ7UyoEphrnbAiz3PGIYiXFwYoADd6o14l1jm/Fj+44tt0eCWw"
+      "/qxF3WLYT8bkrFZyPbmd0GECzGeCIOOFBV+mUbmjyOmtebrmWoLXk6SxfJtWnnCI"
+      "5eQjoVCVqXNsXpzReYK6q8wH+F9C+JUcSf06NPQ2deNzDrF/qVDEhForFESa1MAX"
+      "7Lcn29HHIBaiYqEOissHLXNWSriZf/QJTYPzbU/AADlUGkI6DnzsYxIvlHMoW9u3"
+      "Vuxks9TE1ONISl8x/saVuCfEbK8EALihDm9xW7bD0lQ41jyA19oxrfnZ1TOIFSU6"
+      "1AaXTejHtBVflYZgdBec6KRtuR2aniJ+tqdxfW0XXGJuS5XyDo428/Mosunyrqzh"
+      "HyqFCHGzJbL/GXH63tnVh/gXa71pHrKEb+hLHkoCzvgi6W8aoEP75KDnTd5i/835"
+      "Nkidts36Hwz1VTh6qY+wgUIQVtxC1raI/JvEDqlP3BkydbPw4NoufRm+G/f2/rlh"
+      "cWxb4Bb3sogxvRuMRz7DMfpJl4ndidw+43jU+2PD8NxV9kaBMctjraR8NMFiu/bg"
+      "W7f5b3Nrbb0qhY42eIUIMaK1c2Zz1p0yEYfFmsvAXswYbJBhqZ+dsivs6ZOvYj19"
+      "WaKz8BmCA/9/I6BBCuAkA/XBuWahSqVCNMbgSw6qWdkXSI/oKZw+pl1IUELCTfLJ"
+      "lMOhuUhyhzts7Fw79wLgyzRSw6UG+UUly6UDGL+CcGXQQ6inA6+jghjBBWsRnBvn"
+      "DwgCwtdef9j+52yGxst25MzTTj7EkD0FtG6HA/+WQYMBJJJyYyF9K9haAYA/ln4z"
+      "6f2Uj4Q+aRQjip74YeztIN6tgVPiyeAwb84Y40SmX4kFw8AaPtfP9Z96OcG9zp0V"
+      "aqux2VHrKgfWi0pLkRyDyZpiI1DEJu6B+uTrATtYTcbVLLDGJKXo2NKVxSlk2HtZ"
+      "JqGgSN/+6PBja9j9eGDH5Ujojrlc5L4ZV0x7AdXex2j550AEqk1amkNlkEN/S37j"
+      "y5EXq1rOtAqTf8CabFabZQXIxFkLHB1s3ZY32LwsOOLlL4guBEbIi4hO83hVuo9A"
+      "eMrmvwt7CEAFzNFaRo1IGcICJjm7EctDTeazSB9uTbJCzyhrEiyqylQ1HZhoPQo+"
+      "J7QcRg7dUcZcPFeI15Ros5RNZk8yabyCAX8uewCxP6iAN4nQ2c19zPr4JM9OHioT"
+      "nXuHkBTl8oW8Ibf2FrH66hUmq5MvpoLV+4intUjk25cWV1silx7BrsflRy9YWma5"
+      "JQi+ahqTxcmk7KR6AA1FAW6CHCDl1x6MFrdjAgOIXm1p4tmwqLh8CRmchUgZyW0J"
+      "y8l13r+z8LgfOt/m4suVZGcLnaEFUYOlfDutYc+SXEby2XUSHUlLF/S3CnOY0fww"
+      "KDYYoVYONMFpEYC/j9qMNkm+Av4ynj2IszSH5FdIP8fBn+YOemfbDrPRhDlixwOl"
+      "uZtKvviO/HemZO7FKU4RCOm578y6zTzNOyw5TqF0t1eD5sCjlSXP8aOhJxGEI/ZM"
+      "oW+WOosnnstDM01/+yi0kQv/bWKoZ3aa9fezEBa+tOFT871SJW+s861Sefg1ryAb"
+      "iuDqqZua8O4jFQ=="
+      "-----END ENCRYPTED PRIVATE KEY-----";
+   }
+
+}  // namespace
+
+class Clang_Bug_41810 final : public Test
+   {
+   public:
+      std::vector<Test::Result> run() override
+         {
+         Test::Result result("PKCS8::load_key does not crash when compiled with Clang 8");
+
+         Botan::System_RNG rng;
+
+         auto pem = getPrivateKey();
+         std::string pw = "01234567890123456789"; // sufficiently long password
+
+         Botan::DataSource_Memory ds(reinterpret_cast<const uint8_t*>(pem.data()), pem.size());
+
+         try
+            {
+            Botan::PKCS8::load_key(ds, rng, pw);
+            }
+         catch(const std::exception&) { }
```

Comment:
Should we fail the test if `PKCS8::load_key` reports anything but "wrong password"?

Reply:
Hm, initially I had it fail if there's no exception, but now I think that's not really what we want to test here. So I left it out to avoid confusion. I'm pretty sure there's another test that ensures this already.